test/test_torch.py

```diff
@@ -5940,6 +5940,32 @@ def test_isinf_type(self, device):
         with self.assertRaises(TypeError):
             torch.isinf(1)  # Parameter must be a tensor
 
+    @unittest.skipIf(not TEST_NUMPY, 'NumPy not found')
+    @dtypes(*(torch.testing.get_all_int_dtypes() + torch.testing.get_all_fp_dtypes(include_bfloat16=False)))
+    def test_msort(self, device, dtype):
+        def test(shape):
+            tensor = make_tensor(shape, device, dtype, low=-9, high=9)
+            expected_values = torch.from_numpy(np.msort(tensor.cpu().numpy()))
+            expected_indices = torch.sort(tensor, dim=0).indices
+
+            result_values, result_indices = torch.msort(tensor)
+            self.assertEqual(result_values, expected_values)
+            self.assertEqual(result_indices, expected_indices)
+
+            out = [torch.empty_like(result_values), torch.empty_like(result_indices)]
+            torch.msort(tensor, out=out)
+            self.assertEqual(out[0], expected_values)
+            self.assertEqual(out[1], expected_indices)
+
+        shapes = (
+            [20, 1],
```

Comment:
Add a 1D shape and an empty shape, too.

Reply:
Addressed! But it's seem that `np.msort()` does not support empty shape tensor, so I distinguish the  judging condition of this situation from the general.
```
>>> import torch
>>> import numpy as np
>>> t=torch.empty([])
>>> t
tensor(0.)
>>> torch.msort(t)
tensor(0.)
>>> np.msort(t.cpu().numpy())
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "<__array_function__ internals>", line 5, in msort
  File "C:\Users\Medius Axman\anaconda3\envs\pytorch\lib\site-packages\numpy\lib\function_base.py", line 3374, in msort
    b.sort(0)
numpy.AxisError: axis 0 is out of bounds for array of dimension 0
```

Reply:
Thank you for noticing this. NumPy is inconsistent with its support of empty tensors, and it's great that we support them even when NumPy doesn't.